### PR TITLE
feat(k8s-infra): introduce override config for presets

### PR DESCRIPTION
**Pull request by prashant-shahi**
- For each `k8s-infra` presets, introduce `overideConfig` configuration
- For `kubeletstats` preset, include `metrics` configuration
- Bump up k8s-infra to 0.11.8

Signed-off-by: Prashant Shahi <prashant@signoz.io>

### CODE REVIEW

**Comment by srikanthccv** [comment]
Why not ask people to override rather than providing yet another config option which literally does the override? 

```
resourceDetection:
  enabled: true
  overrideConfig:
    detectors:
      - system
      - env
  timeout: 2s
  override: true
  systemHostnameSources:
    - dns
    - os
  envResourceAttributes: "key=value"
```

Imagine someone seeing something like the above and trying to reason about it. We should not enable people to write such things in the first place.

**Comment by prashant-shahi** [comment]
### TL;DR;

- Either we make sure to support all possible configurations for all presets
- Or we support for overriding the config so that user can include themselves on demand

---

### In Detail

We had a requirement for setting `kubeletstats.metrics` configuration come up - but we didn't way to do that since it was not configured in charts. There would multiple similar ones which would have to be included that might be missed by us.
Having option to override the config could help there for users to include it as per their need.

**Comment by srikanthccv** [comment]
I understand the problem. My point is about how we enable users to solve the problem. Why does every present need to take a new config option which makes the rest of the options useless? And when one wants to customize why should they give a whole section of config (this change is all of the overrideConfig or preset)? Why can't there be a single override that gets merged to the config? This is clean and allows preset or otherwise to be overridden by custom section.

**Comment by prashant-shahi** [comment]
> Why can't there be a single override that gets merged to the config?

Are you suggesting to have single override throughout? We already have the base configs which does that with `otelAgent.config` and `otelDeployment.config`.
Problem is that users would need to update both and it is base config on top of which preset is applied.

> Why does every preset need to take a new config option which makes the rest of the options useless?

That's what the `override` would imply. And only recommended to set if the know complete config of the receiver/processor/exporter.

I understand your concern about the override config adding more confusion to the users.

We could do either of the following:
- Add comments to make it clear on how it works and its impact - retain override config
- Add support for all possible configurations in the component - get rid of override config

**Comment by srikanthccv** [comment]
>We already have the base configs which does that with otelAgent.config and otelDeployment.config.

>it is base config on top of which preset is applied.

This is wrong. The presets config should be merged but the config provided in `otelAgent.config` should get priority.

**Comment by srikanthccv** [comment]
To summerize my point based on my understanding of things, there is a familiar and clean way people override using helm value files. If that can't be used there is a problem with how we did things and should fix that instead of working around that by trying to come up with other solutions which are hard to reason about.

**Comment by prashant-shahi** [comment]
> > We already have the base configs which does that with otelAgent.config and otelDeployment.config.
> 
> > it is base config on top of which preset is applied.
> 
> This is wrong. The presets config should be merged but the config provided in `otelAgent.config` should get priority.

This presets and base config behaviour is inspired from opentelemetry helm charts.
It works great for isolating base config on top of which preset is applied.

Ref: https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-collector/values.yaml#L23

This need not have same behaviour to that of helm override values.

Also, introducing global config override would be same problem as mentioned above. https://github.com/SigNoz/charts/pull/460#issuecomment-2203843821

```yaml
presets:
  resourceDetection:
    enabled: true
    timeout: 2s
    override: true
    systemHostnameSources:
      - dns
      - os
    envResourceAttributes
otelAgent:
  config:
    processors:
      resourcedetection:
        detectors:
          - system
          - env
      ...
```

**Comment by srikanthccv** [comment]
I have a couple of questions


> 
> Also, introducing global config override would be same problem as mentioned above. [#460 (comment)](https://github.com/SigNoz/charts/pull/460#issuecomment-2203843821)
> 
> ```yaml
> presets:
>   resourceDetection:
>     enabled: true
>     timeout: 2s
>     override: true
>     systemHostnameSources:
>       - dns
>       - os
>     envResourceAttributes
> otelAgent:
>   config:
>     processors:
>       resourcedetection:
>         detectors:
>           - system
>           - env
>       ...
> ```

1. Under what circumstances one would do something like this? Even if someone were to do this, what would be the most reasonable thing to do?


2. What is the most familiar approach people use to override config in value files? Should our approach be in the direction that aligns with the larger community or should we introduce our ways of doing things that need further explanation? 

3. Things exist in a certain way as is today for whatever reasons, but if you look at the merits of the choices we have at hand now, what do you think is the best for public chart users?
